app/javascript/packages/document-capture/context/upload.tsx

```diff
@@ -0,0 +1,196 @@
+import { createContext } from 'react';
+import { useObjectMemo } from '@18f/identity-react-hooks';
+import type { ReactNode } from 'react';
+import defaultUpload from '../services/upload';
+
+const UploadContext = createContext({
+  upload: defaultUpload,
+  getStatus: () => Promise.reject() as () => Promise<UploadSuccessResponse>,
+  statusPollInterval: undefined as number | undefined,
+  isMockClient: false,
+  backgroundUploadURLs: {} as Record<string, string>,
+  backgroundUploadEncryptKey: undefined as CryptoKey | undefined,
+  flowPath: 'standard' as FlowPath,
+  csrf: null as string | null,
+});
+
+UploadContext.displayName = 'UploadContext';
+
+type FlowPath = 'standard' | 'hybrid';
+
+/**
+ * Upload field error, after normalized to error instance.
+ */
+export interface UploadFieldError {
+  /**
+   *  Field name
+   */
+  field: 'front' | 'back' | 'selfie' | 'network';
+
+  /**
+   *  Error message.
+   */
+  message: string;
+}
+
+interface UploadOptions {
+  /**
+   * HTTP method to send payload.
+   */
+  method?: 'POST' | 'PUT';
+
+  /**
+   *  Endpoint to which payload should be sent.
+   */
+  endpoint: string;
+
+  /**
+   *  CSRF token to send as parameter to upload implementation.
```

Comment:
Nit: Some extra spaces

```suggestion
   * Field name
   */
  field: 'front' | 'back' | 'selfie' | 'network';

  /**
   * Error message.
   */
  message: string;
}

interface UploadOptions {
  /**
   * HTTP method to send payload.
   */
  method?: 'POST' | 'PUT';

  /**
   * Endpoint to which payload should be sent.
   */
  endpoint: string;

  /**
   * CSRF token to send as parameter to upload implementation.
```